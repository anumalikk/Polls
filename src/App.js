import "./App.css";
import "./index.css";
import React from "react";
import Nav from "./components/nav";
import Dashboard from "./components/Dashboard";
import { Route, Routes, useLocation } from "react-router-dom";
import QuestionDetails from "./components/QuestionDetails";
import LeaderBoard from "./components/LeaderBoard";
import NewPoll from "./components/NewPoll";
import Login from "./components/Login";
import PrivateRoute from "./components/PrivateRoute";
import PageNotFound from "./components/PageNotFound";

function App() {
  const location = useLocation();

  // Conditionally render the <Nav /> component based on the route
  const renderNav = location.pathname !== "/login";

  return (
    <div className="App">
      {renderNav && <Nav />}

      <Routes>
        <Route path="/login" exact element={<Login />} />
        <Route
          path="/"
          element={
            <PrivateRoute>
              <Dashboard />
            </PrivateRoute>
          }
        />
        <Route
          path="/leaderboard"
          exact
          element={
            <PrivateRoute>
              <LeaderBoard />
            </PrivateRoute>
          }
        />
        <Route
          path="/questions/:id"
          element={
            <PrivateRoute>
              <QuestionDetails />
            </PrivateRoute>
          }
        />
        <Route
          path="/new"
          exact
          element={
            <PrivateRoute>
              <NewPoll />
            </PrivateRoute>
          }
        />
        <Route path="/404" exact element={<PageNotFound />} />
      </Routes>
    </div>
  );
}



export default App;
