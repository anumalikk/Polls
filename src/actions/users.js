export const RECEIVE_USERS = "RECEIVE_USERS";
export const RESET_USERS = "RESET_USERS";
export const ADD_ANSWER_USER = "ADD_ANSWER_USER";
export const ADD_USER_QUESTION = "ADD_USER_QUESTION";

export function receiveUsers(users) {
  return {
    type: RECEIVE_USERS,
    users,
  };
}

export function addAnswerUser(authedUser, qid, answer) {
  return {
      type: ADD_ANSWER_USER,
      authedUser,
      qid,
      answer,
  };
}

export function resetUsers() {
  return {
    type: RESET_USERS,
    
  };
}

export function addUserQuestion({author, id}) {
  return {
      type: ADD_USER_QUESTION,
      author,
      qid: id,
  };
}