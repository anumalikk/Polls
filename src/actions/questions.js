import { saveQuestion, saveQuestionAnswer } from "../utils/api";
import { addAnswerUser, addUserQuestion } from "./users";

export const RECEIVE_QUESTIONS = "RECEIVE_QUESTIONS";
export const RESET_QUESTIONS = "RESET_QUESTIONS";
export const ADD_ANSWER_QUESTION = "ADD_ANSWER_QUESTION";
export const ADD_QUESTION = "ADD_QUESTION";

export function receiveQuestions(questions) {
  return {
    type: RECEIVE_QUESTIONS,
    questions,
  };
}

export function resetQuestions() {
  return {
    type: RESET_QUESTIONS,
  };
}

function addAnswerQuestion(author, qid, answer) {
  return {
    type: ADD_ANSWER_QUESTION,
    author,
    qid,
    answer,
  };
}

function addQuestion(question) {
  return {
    type: ADD_QUESTION,
    question,
  };
}

export function handleAddAnswer(authedUser, questionId, answer) {
  return (dispatch) => {
    return saveQuestionAnswer(authedUser, questionId, answer).then(() => {
      dispatch(addAnswerQuestion(authedUser, questionId, answer));

      dispatch(addAnswerUser(authedUser, questionId, answer));
    });
  };
}

export function handleAddQuestion(question) {
  return (dispatch) => {
    return saveQuestion(question).then((question) => {
      
      dispatch(addQuestion(question));

      dispatch(addUserQuestion(question));
    });
  };
}
