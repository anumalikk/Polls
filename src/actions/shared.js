import { getInitialData } from "../utils/api";
import { receiveUsers, resetUsers } from "./users";
import { receiveQuestions, resetQuestions } from "./questions";
import { setAuthedUser, logoutAuthedUser } from "./authedUser";

export function handleInitialDataForLogin(authedUser) {
  return (dispatch) => {    
    return getInitialData().then(({ users, questions }) => {
      dispatch(receiveUsers(users));
      console.log(users);
      console.log(questions);      
      dispatch(receiveQuestions(questions));      
      dispatch(setAuthedUser(authedUser));      
    });
  };
}

export function handleLogout() {
  return (dispatch) => {
    dispatch(logoutAuthedUser());
    dispatch(resetQuestions());
    dispatch(resetUsers());
  };
}

