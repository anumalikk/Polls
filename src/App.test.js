import React from 'react';
import {render} from '@testing-library/react';
import App from './App';
import {Provider} from "react-redux";
import {store} from "./store";
// import {store} from "./index";
import {BrowserRouter} from "react-router-dom";
import {setAuthedUser} from "./actions/authedUser";

describe("App", () => {
    it("This test case would render the App component all well", () => {
        // render(<App></App>)
        const component = render(
            <Provider store={store}>
                <BrowserRouter>
                    <App/>
                </BrowserRouter>
            </Provider>
        );
        expect(component).toBeDefined();
        expect(component).toMatchSnapshot();
    });

    it("This test case would show Login page when user has not logged in using login page which contains list of users", () => {
        const component = render(
            <Provider store={store}>
                <BrowserRouter>
                    <App/>
                </BrowserRouter>
            </Provider>
        );
        const heading = component.getByTestId("login-h1");
        expect(heading).toBeInTheDocument();
    });

    it("This test case would show Dashboard page once user is logged in successfully using login page", () => {
        store.dispatch(setAuthedUser("sarahedo"));

        const component = render(
            <Provider store={store}>
                <BrowserRouter>
                    <App/>
                </BrowserRouter>
            </Provider>
        );

        const heading = component.getByTestId("h1-heading");
        expect(heading).toBeInTheDocument();
    });
});
