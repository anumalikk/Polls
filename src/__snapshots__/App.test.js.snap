// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App This test case would render the App component all well 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="App"
      >
        <div>
          <h1
            class="text-3xl font-bold mt-9"
            data-testid="login-h1"
          >
            Login
          </h1>
          <form>
            <div>
              <label>
                Select a user:
                <select>
                  <option
                    value=""
                  >
                    Select a user
                  </option>
                  <option
                    value="sarahedo"
                  >
                    sarahedo
                  </option>
                  <option
                    value="tylermcginnis"
                  >
                    tylermcginnis
                  </option>
                  <option
                    value="mtsamis"
                  >
                    mtsamis
                  </option>
                  <option
                    value="zoshikanlu"
                  >
                    zoshikanlu
                  </option>
                </select>
              </label>
            </div>
            <div
              class="mt-6 text-right"
            >
              <button
                class="bg-sky-500 hover:bg-sky-700 px-5 py-2.5 text-sm leading-5 rounded-md font-semibold text-white"
                data-testid="submit"
                type="submit"
              >
                Login
              </button>
            </div>
          </form>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="App"
    >
      <div>
        <h1
          class="text-3xl font-bold mt-9"
          data-testid="login-h1"
        >
          Login
        </h1>
        <form>
          <div>
            <label>
              Select a user:
              <select>
                <option
                  value=""
                >
                  Select a user
                </option>
                <option
                  value="sarahedo"
                >
                  sarahedo
                </option>
                <option
                  value="tylermcginnis"
                >
                  tylermcginnis
                </option>
                <option
                  value="mtsamis"
                >
                  mtsamis
                </option>
                <option
                  value="zoshikanlu"
                >
                  zoshikanlu
                </option>
              </select>
            </label>
          </div>
          <div
            class="mt-6 text-right"
          >
            <button
              class="bg-sky-500 hover:bg-sky-700 px-5 py-2.5 text-sm leading-5 rounded-md font-semibold text-white"
              data-testid="submit"
              type="submit"
            >
              Login
            </button>
          </div>
        </form>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
